cfme/tests/cloud/test_providers.py

```diff
@@ -383,59 +384,67 @@ def test_create_arbitration_profiles(self, rest_api, arbitration_profiles):
         Metadata:
             test_flag: rest
         """
-        assert len(arbitration_profiles) > 0
-        record = rest_api.collections.arbitration_profiles.get(id=arbitration_profiles[0].id)
-        assert record._data == arbitration_profiles[0]._data
-        assert 'ArbitrationProfile' in arbitration_profiles[0].type
+        for profile in arbitration_profiles:
+            record = rest_api.collections.arbitration_profiles.get(id=profile.id)
+            assert rest_api.response.status_code == 200
+            assert record._data == profile._data
+            assert 'ArbitrationProfile' in profile.type
 
     @pytest.mark.tier(3)
     @pytest.mark.uncollectif(lambda: version.current_version() < '5.7')
-    @pytest.mark.parametrize(
-        "from_detail", [True, False],
-        ids=["from_detail", "from_collection"])
-    def test_delete_arbitration_profiles(self, rest_api, arbitration_profiles, from_detail):
-        """Tests delete arbitration profiles.
+    @pytest.mark.parametrize('method', ['post', 'delete'])
+    def test_delete_arbitration_profiles_from_detail(self, rest_api, arbitration_profiles, method):
+        """Tests delete arbitration profiles from detail.
+
+        Metadata:
+            test_flag: rest
+        """
+        status = 204 if method == 'delete' else 200
+        for entity in arbitration_profiles:
+            entity.action.delete(force_method=method)
+            assert rest_api.response.status_code == status
+            with error.expected('ActiveRecord::RecordNotFound'):
+                entity.action.delete(force_method=method)
+            assert rest_api.response.status_code == 404
+
+    @pytest.mark.tier(3)
+    @pytest.mark.uncollectif(lambda: version.current_version() < '5.7')
+    def test_delete_arbitration_profiles_from_collection(self, rest_api, arbitration_profiles):
+        """Tests delete arbitration profiles from collection.
 
         Metadata:
             test_flag: rest
         """
-        assert len(arbitration_profiles) > 0
         collection = rest_api.collections.arbitration_profiles
-        if from_detail:
-            methods = ['post', 'delete']
-            for i, ent in enumerate(arbitration_profiles):
-                ent.action.delete(force_method=methods[i % 2])
-                with error.expected("ActiveRecord::RecordNotFound"):
-                    ent.action.delete()
-        else:
+        collection.action.delete(*arbitration_profiles)
```

Comment:
**Optional**
This actually seems like two tests to me - one for deleting a single item from the collection, and one for deleting multiple items from the collection.

Can definitely be a TODO or something like that and doesn't have to be done in this PR.

I see you've moved the assert against the length of arbitration_profiles into the fixture, and it all plays together. I'm thinking it may be best to have a soft_assert in the fixture if the number of created doesn't match the intended, and then keep a hard assert inside the cases where the test won't actually assert/do anything if the list is empty.